File: Loenn/triggers/bomb_timer_trigger.lua
local bombTimer = {}

bombTimer.name = "vitellary/bombtimer"

bombTimer.fieldInformation = {
    startDirection = {
        editable = false,
        options = {
            "Any",
            "Right",
            "Left",
            "Up",
            "Down",
        },
    },
    timer = {
        minimumValue = 0,
    },
    soundAt = {
        minimumValue = 0,
        maximumValue = function(room, entity)
            return entity.timer
        end,
    },
}

bombTimer.placements = {
    {
        name = "bomb_timer",
        data = {
            sound = "",
            soundAt = 0,
            timer = 0,
            startDirection = "Any",
            changeRespawn = true,
            resetOnDeath = true,
        }
    }
}

return bombTimer

File: Ahorn/triggers/bombTimerTrigger.jl
module FlushelineBombTimerTrigger

using ..Ahorn, Maple

@mapdef Trigger "vitellary/bombtimer" BombTimerTrigger(x::Integer, y::Integer, width::Integer=16, height::Integer=16, sound::String="", soundAt::Number=0.0, timer::Number=0.0, startDirection::String="Any", changeRespawn::Bool=true, resetOnDeath::Bool=true)

const placements = Ahorn.PlacementDict(
    "Bomb Timer Trigger (Crystalline)" => Ahorn.EntityPlacement(
        BombTimerTrigger,
        "rectangle"
    )
)

const startDirections = String["Any", "Right", "Left", "Up", "Down"]

Ahorn.editingOptions(trigger::BombTimerTrigger) = Dict{String, Any}(
    "startDirection" => startDirections
)

end